src/mjolnir/landmark_database_builder.cc

```diff
@@ -169,10 +200,16 @@ std::vector<Landmark> LandmarkDatabase::get_landmarks_in_bounding_box(const doub
   int ret = sqlite3_step(bounding_box_stmt);
   while (ret == SQLITE_ROW) {
     const char* name = reinterpret_cast<const char*>(sqlite3_column_text(bounding_box_stmt, 0));
-    const char* type = reinterpret_cast<const char*>(sqlite3_column_text(bounding_box_stmt, 1));
+
+    LandmarkType type = LandmarkType::NA;
+    if (sqlite3_column_type(bounding_box_stmt, 1) != SQLITE_NULL) {
+      type = static_cast<LandmarkType>(sqlite3_column_int(bounding_box_stmt, 1));
+    }
+
```

Comment:
I updated landmark type to enum of uint8_t which introduces some data type conversions like here.
I wonder if this could be a problem.